TypeCobol/Compiler/CupParser/NodeBuilder/ProgramClassBuilder.cs

```diff
@@ -1341,6 +1341,22 @@ public virtual void StartWhenConditionClause(List<TypeCobol.Compiler.CodeElement
 
         public virtual void EndWhenConditionClause()
         {
+            if (SyntaxTree.CurrentNode.ChildrenCount == 0)
+            {
+                System.Diagnostics.Debug.Assert(SyntaxTree.CurrentNode.Parent.ChildrenCount >= 2);
+                // -1 is the Then node
+                // -2 is the When group
+                var whenGroup = SyntaxTree.CurrentNode.Parent.Children[SyntaxTree.CurrentNode.Parent.ChildrenCount - 2];
+                System.Diagnostics.Debug.Assert(whenGroup.ChildrenCount > 0);
+
+                var whenNode = whenGroup.Children[whenGroup.ChildrenCount - 1];
+                System.Diagnostics.Debug.Assert(whenNode.CodeElement != null);
+                System.Diagnostics.Debug.Assert(whenNode.CodeElement.Type == CodeElementType.WhenCondition ||
+                    whenNode.CodeElement.Type == CodeElementType.WhenSearchCondition);
+
+                //Syntax error.
+                DiagnosticUtils.AddError(whenNode, "Missing statement in \"when\" clause");
```

Comment:
Diagnostics are usually created in checkers. Can we move that ?

Reply:
But here  it's a syntactical error, why move it in crosschecker, I think crosschecker is used for semantic errors ?

Reply:
Same remarks than CodeElementBuilder, all diagnostics are created in a checker.